### DISCUSSION TITLE
Docker Container doesn't start after update to 1.25.2

### DISCUSSION THREAD

**Discussion by esackbauer** [dani-garcia/vaultwarden#2656] [Q&A] [answered]
### Deployment environment

* vaultwarden version:

* Install method:
Docker / Portainer 2.14

* Clients used: 

* Reverse proxy and version: 

* MySQL/MariaDB or PostgreSQL version: 

* Other relevant details:

### Steps to reproduce

Starting the container after update to VaultWarden 1.25.2 it says there is no persistent volume. But the volume is actually mounted:
[INFO] No .env file found.

[DEPRECATED]: `SMTP_SSL` or `SMTP_EXPLICIT_TLS` is set. Please use `SMTP_SECURITY` instead.

[2022-07-28 08:17:09.244][vaultwarden][ERROR] No persistent volume!

########################################################################################

# It looks like you did not configure a persistent volume!                             #

# This will result in permanent data loss when the container is removed or updated!    #

# If you really want to use volatile storage set `I_REALLY_WANT_VOLATILE_STORAGE=true` #

#######################################################################################
### Troubleshooting data

Contents of dockers volume:
}root@app03:/var/lib/docker/volumes/1d24e6fe4517c3abe615941fdc2ec436a14900fed768d6ebdb3397a1cc0a1926/_data# ls -la
total 2808
drwxr-xr-x 6 root root    4096 Jul 17 13:22 .
drwxr-xr-x 3 root root    4096 Jan 30  2020 ..
drwxr-xr-x 5 root root    4096 Feb 10  2020 attachments
-rw-r--r-- 1 root root     109 Jan 31  2020 config.json
-rw-r--r-- 1 root root 1392640 Jul 17 13:22 db.sqlite3
-rw-r--r-- 1 root root   32768 Jul 28 09:43 db.sqlite3-shm
-rw-r--r-- 1 root root 1376112 Jul 28 09:43 db.sqlite3-wal
drwxr-xr-x 2 root root   20480 Jul 28 09:37 icon_cache
-rw------- 1 root root    1192 Jan 30  2020 rsa_key.der
-rw------- 1 root root    1679 Jan 30  2020 rsa_key.pem
-rw-r--r-- 1 root root     270 Jan 30  2020 rsa_key.pub.der
-rw-r--r-- 1 root root     451 Jun 28  2021 rsa_key.pub.pem
drwxr-xr-x 2 root root    4096 May 24 18:50 sends
drwxr-xr-x 2 root root    4096 May 24 18:50 tmp

**Comment by esackbauer** [comment]
Switching back to previous image version 1.25.1 and the container starts and works.

**Comment by BlackDex** [comment, marked as answer]
Yes, but it looks like that isn't a persistent volume.
A persistent volume has a different path.

What that means, is that if you now remove this container, docker will also remove that volume, since it is not persistent.
If you would create a volume, and link that then docker will not remove that volume, because it is defined separately.

So, either create a volume or bind-mount a host directory and copy all those files, or life on the edge and configure that variable.

**Comment by esackbauer** [comment]
It is a persistent volume. It is not removed when I remove the container. I am using Portainer, so it might look like its not a persistent volume, but it is.
When I downgraded to version 1.25.1, all data was instantly available, with that same volume.

**Comment by BlackDex** [comment]
I'm afraid it isn't an actual persistent volume.

A persistent volume is configured like this:
![image](https://user-images.githubusercontent.com/610450/181500865-963acd0d-5093-44c2-a3ff-b532ad29585d.png)

If i remove a non-persistent vaultwarden, i do receive this message:
![image](https://user-images.githubusercontent.com/610450/181501233-8a174584-00a5-48fb-bef6-04087e422e0a.png)

While portainer seems to give you an option, and also seems to have it disabled by default, it technically is a non-persistent volume. Because, if i enable that option it removes that volume of the non-persistent container. If i enable that same option on the persistent container, the volume is still there.

So, this is not a persistent volume, but an ephemeral/non-persistent one.

**Comment by BlackDex** [comment]
Not sure what happend, your comment was removed for some reason, either by your self, or something else, but i'm not seeing it anymore.
Still, here is my comment on your post:

_I left out the screenshots btw_
> It is configured as persistent volume, as you have shown. Here are my screenshots:
> It just has a random name. But it IS a persistent volume.

Well. Then i suggest that you enable the `I_REALLY_WANT_VOLATILE_STORAGE=true` option in this case.
And, to be fair. If you would have created the volume your self manually and give it a name, it wouldn't have this format.
It looks like this is was a non-persistent volume before, and later used as a persistent volume.

So, since you know what you are doing, i suggest to enable that flag, and start using the latest version.
We added this option because sometimes people do forget to create a volume or use a bind mount.

So, this is as intended. But if you know of any other way to verify that this is a persistent volume I'm all ears on implementing that.

**Reply by esackbauer** [reply]
I deleted the comment because in the meantime I created a new "persistant" volume, copied all data over from the "non-persistent" volume. And voila, 1.25.2 starts as usual.

Sorry for the inconvenience.

**Reply by BlackDex** [reply]
No, no problem. It's better to have these topics here so it can be helpful to other people with the same issue.

**Reply by 3urobeat** [reply]
Absolutely. You just saved me a lot of trouble ^^  
Creating a volume manually, giving it a 'normal' name, copying files over and recreating the vaultwarden container with the new volume solved it for me as well. The other volume was persistent as well and mounted the same way, I guess vaultwarden just doesn't want to accept that when it has an automatic generated name?

**Reply by BlackDex** [reply]
@HerrEurobeat if you are also using Portainer, then they really are non-persistent volumes, just read the stuff above again.
And if for some reason auto-generated volumes have those names, then yes, we will not except them, as they do represent non-persistent volumes, and there were people who already lost there data because of this. This is a feature to help and prevent those losses. If you are really certain it isn't then just add the `I_REALLY_WANT_VOLATILE_STORAGE=true` and be done with it.

**Comment by Jack0000000000000** [comment]
Hi, I have the same problem. WHERE exactly can I change it? In which file? I am new to dockers.
